### ISSUE TITLE
Rule `PLR6301` ignore comment requires odd location

### ISSUE THREAD

**Issue by ofek** [astral-sh/ruff#8635]
```
Method `construct_build_command` could be a function, class method, or static method
```

The following is the proper way to ignore `PLR6301` when the function/method spans multiple lines:

```python
def construct_build_command(
    self,  # noqa: PLR6301
    *,
    directory=None,
    targets=(),
    hooks_only=False,
    no_hooks=False,
    clean=False,
    clean_hooks_after=False,
    clean_only=False,
):
```

Should it not be the case that ignore comments be on the method?

**Comment by charliermarsh** [comment]
Yeah this seems off. I think it's flagging the `self`, but we should probably move it to the method name.

**Comment by ofek** [comment]
Thanks!
